soh/soh/Enhancements/mods.cpp

```diff
@@ -953,6 +953,23 @@ void RegisterAltTrapTypes() {
     });
 }
 
+void RegisterRandomizerSheikSpawn() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnSceneSpawnActors>([]() {
+        if (!gPlayState) return;
+        bool canSheik = OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_TRIAL_COUNT) != RO_GANONS_TRIALS_SKIP;
+        if (!gSaveContext.n64ddFlag || !LINK_IS_ADULT || !canSheik) return;
+        switch (gPlayState->sceneNum) {
+            case SCENE_TEMPLE_OF_TIME:
```

Comment:
You are only looking at the scene, and not the room that you want to spawn Sheik in. This means that every room transition e.g. Ganon's tower (going to all the trials) is going to spawn Sheik, even though you cant get to her.

You should be checking for room here too before spawning.